### PR TITLE
Hd abhishek

### PR DESCRIPTION
**Pull request by gem-abhishek-gautam**

### CODE REVIEW

File: src/test/resources/features/CreateTicket.feature [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    And Upload file with invalid extensions and size
+    Then Verify if "invalid" file can be uploaded
+
+  Scenario: Check for field validation in create ticket form
+    Given Open create ticket form
+    Then Verify subject character limit and required field values
+
+  Scenario: Check comments and files can be added for incident ticket details
+    Given Create a ticket for incident with subject "incident test", desc "desc test" and dept "Accounts"
+    When Submit and verify the incident details having file "unattached", subject "incident test", dept "Accounts" and description "desc test"
```
Comment (Gem-NeerajSingh) [inline src/test/resources/features/CreateTicket.feature:162]: Then---When Submit and verify

File: src/test/resources/features/CreateTicket.feature [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    Then Verify if comment "no file" can be added for the ticket
+    Then Validate comment with file uploads on ticket page
+
+  Scenario: Check comments and files can be added for request ticket details
+    Given Create request without file and with subject "request test", dept "HR", category "Leave Management", sub-category "LNSA" and description "desc test"
+    When Submit and verify the request details having file "unattached", subject "request test", dept "HR", description "desc test", category "Leave Management" and sub-category "LNSA"
```
Comment (Gem-NeerajSingh) [inline src/test/resources/features/CreateTicket.feature:168]: then----When Submit and verify

**Comment by gem-abhishek-gautam** [comment]
Changes done

File: src/test/java/com/qa/helpdesk/stepDefinitions/SupportTicketStepDefinition.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            GemTestReporter.addTestStep("Exception Occurred", "Exception: " + e, STATUS.FAIL);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Then("Submit and verify ticket with file {string} and type {string}, caller {string}, channel {string}, dept {string}, category {string}, sub-category {string}, subject {string}, desc {string} and status {string}")
+    public void submitAndVerifyTicketWithFileAndTypeCallerSelf(String fileFlag, String type, String caller, String channel, String dept, String category, String subcategory, String subject, String desc, String status) {
+
+        try {
+            //TODO
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/SupportTicketStepDefinition.java:36]: todo????????????

File: src/test/java/com/qa/helpdesk/stepDefinitions/SupportTicketStepDefinition.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        } catch (Exception e) {
+            GemTestReporter.addTestStep("Exception Occurred", "Exception: " + e, STATUS.FAIL);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @When("Caller name {string} is provided")
+    public void callerNameIsProvided(String name) {
+        try {
+            //TODO
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepDefinitions/SupportTicketStepDefinition.java:46]: Unimplemented block

Comment (gem-abhishek-gautam) [reply src/test/java/com/qa/helpdesk/stepDefinitions/SupportTicketStepDefinition.java:46]: Will implement them

Comment (gem-abhishek-gautam) [reply src/test/java/com/qa/helpdesk/stepDefinitions/SupportTicketStepDefinition.java:46]: And added to next PR after implementation

**Pull request closed without merging**